src/courses/views.py

```diff
@@ -261,6 +261,22 @@ class BlockDelete(NonPublicOnlyViewMixin, DeleteView):
     model = Block
     success_url = reverse_lazy('courses:block_list')
 
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        course_student_qs = self.get_object().coursestudent_set
+        context["population"] = course_student_qs.count()
+        context['populated'] = course_student_qs.exists()
+        return context
+
+    def post(self, request, *args, **kwargs):
+        
+        populated = self.get_object().coursestudent_set.exists()
+        if populated:
+            return self.get(request, *args, **kwargs)
+
+        return super().post(request, *args, **kwargs)
+
```

Comment:
It's not obvious to me what you're doing here, can you add a code comment explaining the purpose of this?

Reply:
done